cloud-functions/functions/__tests__/auth.test.ts

```diff
@@ -56,6 +53,28 @@ afterEach(async () => {
   }
 });
 
+test('An attacker cannot create an account and then quickly create their own record in /users table before onCreate has time to run and delete them', async () => {
```

Comment:
"before onCreate has time to run and delete them" is confusing. Why does the record have to be deleted before we can create it?

Reply:
`onCreate` is a cloud function that's called whenever a user is created in Firebase Auth, that we're using in part to delete any users that register with Firebase Auth but aren't in our `/users` collection. That way, public sign ups are effectively "turned off", even though Firebase Auth doesn't support that as a feature.

This is a simple attack I thought of to try and circumvent that system. Maybe the test could be worded better, though I think in context of understanding our system it makes sense (?)